src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -2,3 +2,231 @@
+def db_create_secondary(
+    client,
+    database_name,
+    server_name,
+    resource_group_name,
+    secondary_server_name,
+    secondary_resource_group_name=None,
+    **kwargs):
+
+    # Determine optional values
+    secondary_resource_group_name = secondary_resource_group_name or resource_group_name
+
+    # Set create mode
+    kwargs['create_mode'] = 'OnlineSecondary'
+
+    return _db_create_special(
+        client,
+        database_name,
+        server_name,
+        resource_group_name,
+        database_name, # replica must have the same database name as the source db
+        secondary_server_name,
+        secondary_resource_group_name,
+        kwargs)
+
+# Creates a database from a database point in time backup. Wrapper function to make create mode more convenient.
+#def db_restore(
+#    client,
+#    database_name,
+#    server_name,
+#    resource_group_name,
+#    restore_point_in_time,
+#    dest_name,
+#    dest_server_name=None,
+#    dest_resource_group_name=None,
+#    **kwargs):
+
+#    # Determine optional values
+#    dest_resource_group_name = dest_resource_group_name or resource_group_name
+#    dest_server_name = dest_server_name or server_name
+
+#    # Set create mode: TODO - doesn't yet work because restorePointInTime is not yet in Swagger spec
+#    kwargs['create_mode'] = 'PointInTimeRestore'
+#    #kwargs['restore_point_in_time'] = restore_point_in_time
+
+#    return _db_create_special(
+#        client,
+#        database_name,
+#        server_name,
+#        resource_group_name,
+#        dest_name,
+#        dest_server_name,
+#        dest_resource_group_name,
+#        kwargs)
+
+# Lists databases in a server or elastic pool.
+def db_list(
+    client,
+    server_name,
+    resource_group_name,
+    elastic_pool_name=None):
+
+    if elastic_pool_name:
+        # List all databases in the elastic pool
+        pool_client = get_sql_elasticpools_operations(None)
+        return pool_client.list_databases(
+            server_name=server_name,
+            resource_group_name=resource_group_name,
+            elastic_pool_name=elastic_pool_name)
+        return 
+    else:
+        # List all databases in the server
+        return client.list_by_server(
+            resource_group_name=resource_group_name,
+            server_name=server_name)
+
+###############################################
+#                sql elastic-pool             #
+###############################################
+
+# Creates an elastic pool. Wrapper function which uses the server location so that the user doesn't
+# need to specify location.
+def elastic_pool_create(
+    client,
+    server_name,
+    resource_group_name,
+    elastic_pool_name,
+    **kwargs):
+
+    # Determine server location
+    kwargs['location'] = get_server_location(server_name=server_name, resource_group_name=resource_group_name)
+
+    # Create
+    return client.create_or_update(
```

Comment:
Should we first validate that the elastic pool doesn't already exist? That way the customer doesn't accidently update an existing elastic pool rather than creating a new one.